### PR TITLE
Use the official Heroku buildpack

### PR DESCRIPTION
**Pull request by chrisvogt**
See my comment on #50. Also see comments at this [project issue](https://github.com/CHH/heroku-buildpack-php/issues/165). 

The changes in this PR make this app template work out of the box again. This also updates the README and `composer.json`, and can replace the unmerged, behind branch that @josegonzalez started on last June.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,12 +19,5 @@
   "minimum-stability": "dev",
   "config": {
     "preferred-install": "dist"
-  },
-  "extra": {
-    "heroku": {
-      "framework": "cakephp2",
-      "document-root": "webroot",
-      "index-document": "index.php"
-    }
```
Comment (josegonzalez) [inline composer.json]: Does keeping this break anything?

Comment (chrisvogt) [inline composer.json]: Everything appears to be working fine on a test I'm running with Cake v2.6.0. That appears to be specific to the CHH buildpack, and is instead now defined in a Procfile.

Comment (josegonzalez) [inline composer.json]: Can you keep that section in for now?

**Comment by chrisvogt** [comment]
How does that look @josegonzalez? I added back the `extra` object but without `framework` set since that one is [definitely CHH-specific](https://github.com/CHH/heroku-buildpack-php/blob/master/frameworks/cakephp2). I can add it back at your call.

**Comment by josegonzalez** [comment]
Can you add that back as well? Thanks in advance :)

**Comment by chrisvogt** [comment]
Sure thing, done. :dancer: 

What are your concerns here -- backwards compatibility?

**Comment by AD7six** [comment]
@chrisvogt I'd say so. Only adding things means it's a minor/patch release rather than a major. Can't be sure no-one is using stuff for other purposes ([troll-cough](https://github.com/tomzx/php-semver-checker/commit/1a0ba465c1d520c5b314257681b1e5e412dc997d#diff-04c6e90faac2675aa89e2176d2eec7d8R31)).

**Comment by chrisvogt** [comment]
:ok: I see, thank you.

**Pull request merged**